### PR TITLE
Document usage with PgBoucner

### PR DESCRIPTION
**Pull request by 2color**

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx [hunk 1]
```diff
@@ -74,18 +74,45 @@ postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
...
 
+### Usage with PgBouncer
+
+Typically PostgreSQL supports only a certain amount of concurrent connections and this limit can be reached quite fast when the service usage goes up – especially in serverless environments.
+
+A connection pooler like PgBouncer prevents your application from exhausing the database's connection limit.
+
+PgBouncer holds a connection pool to the database and proxies incoming client connections by sitting between Prisma Client and the database.
+
+PgBouncer funnels client connections into a pooling application, which reduces the number of processes a database has to handle at any given time. PgBouncer passes on a limited number of connections to the database and queues additional connections for delivery when space becomes available.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/190-deployment.mdx]: First half of the first sentence says the same as the previous paragraph.

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx [hunk 2]
```diff
@@ -74,18 +74,45 @@ postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
...
+
+To use Prisma Client with PgBouncer from a serverless function, add the `?pgbouncer=true` flag to the PostgreSQL connection string and set the connection limit to `1`:
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/190-deployment.mdx]: Setting Connection Limit to 1 is unrelated to usage of PgBouncer. It is a thing you can do in specific circumstances (when you are running on serverless with an application that never uses more than 1 connectino in parallel or you really want to optimize on the number of serverless functions that can run in parallal)

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx [hunk 3]
```diff
@@ -74,18 +74,45 @@ postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
...
+```
+
+Additionally, for Prisma Client to work reliably, PgBouncer must run in [**Transaction mode**](https://www.pgbouncer.org/features.html).
+
+_Transaction mode_ offers a connection for every transaction. With this mode in addition to the , the Prisma query engine is able to operate normally.
+
+<details>
+  <summary>How `pgbouncer` mode works in Prisma</summary>
+- Prisma will clean up already present prepared statements in the connection by running `DEALLOCATE ALL` before trying to prepare and execute the Prisma Client query.
+- Prisma will triggers a transaction for every query case –even when just reading data– which allows Prisma to use prepared statements.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/190-deployment.mdx]:
```suggestion
- Prisma will trigger a transaction for every query case – even when just reading data – which allows Prisma to use prepared statements.
```

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx [hunk 4]
```diff
@@ -74,18 +74,43 @@ postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
 mysql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
 ```
 
+### Usage with PgBouncer
+
+Typically PostgreSQL supports only a certain amount of concurrent connections and this limit can be reached quite fast when the service usage goes up – especially in serverless environments.
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/190-deployment.mdx]: Typically seems unecessary here?

```suggestion
PostgreSQL supports only a certain amount of concurrent connections and this limit can be reached quite fast when the service usage goes up – especially in serverless environments.
```

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx [hunk 5]
```diff
@@ -74,18 +74,43 @@ postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
...
+
+To use Prisma Client with PgBouncer from a serverless function, add the `?pgbouncer=true` flag to the PostgreSQL connection string:
+
+```
+postgresql://USER:PASSWORD@HOST:PORT/DATABASE?pgbouncer=true
+```
+
+Additionally, for Prisma Client to work reliably, PgBouncer must run in [**Transaction mode**](https://www.pgbouncer.org/features.html).
+
+_Transaction mode_ offers a connection for every transaction. With this mode in addition to the , the Prisma query engine is able to operate normally.
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/190-deployment.mdx]: seems like this sentence is not complete? "in addition to the ..."?

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx [hunk 6]
```diff
@@ -74,18 +74,43 @@ postgresql://USER:PASSWORD@HOST:PORT/DATABASE?connection_limit=1
...
+
+<details>
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/190-deployment.mdx:95]: Seems like the content in here doesn't render properly as Markdown right now:

image

Maybe try to add new lines before and after the bullet list? If that doesn't work, we should summon Nilu to fix the rendering.

**Review by nikolasburk — CHANGES_REQUESTED** [review]
This looks great Daniel 👏 I've left a few comments but overall I think this will be really helpful content for a lot of folks 🙂

File: content/200-concepts/100-components/02-prisma-client/190-deployment.mdx (hunk 6 above)
Comment (2color) [reply content/200-concepts/100-components/02-prisma-client/190-deployment.mdx:95]: Yep, this was a whitespace problem. Adding one before and after the bullet list fixed it:
Screen Shot 2021-01-26 at 18 53 58

**Pull request merged**